### ISSUE TITLE
Add support for more Maclaurin Series models

### ISSUE THREAD

**Issue by HarveyRDonnelly** [HarveyRDonnelly/maclaurin-series-models#1]
By using the existing mathematical models as a template, I think it would be beneficial to add some further models which illustrate more interesting Maclaurin Series approximations for different functions. For example, building support for hyperbolic trigonometric functions, inverse tan x, geometric series etc.

**Comment by ckmganesh** [comment]
Hey, there. I'm not familiar with the maths behind this. Can you guide me with some more information regarding what to do

**Comment by HarveyRDonnelly** [comment]
Thanks for your interest in this project!

Firstly, a Taylor Series of a mathematical function is an infinite sum of polynomials (i.e. a + bx + cx<sup>2</sup> + ... +  dx<sup>n</sup>) with which the coefficients of each term are calculated from the derivatives of that function at a given point. A Maclaurin Series is simply a Taylor Series where this given point is zero. As the number of terms in this sum approaches infinity, the polynomial function serves as an increasingly accurate approximation of the initial function.

As we add more terms and increase the degree of the polynomial, the polynomial function will become an increasingly closer approximation of the original function. This concept is precisely what this library's mathematical models aim to visualise.

[This Khan Academy video may help with learning how to calculate the coefficients.](https://youtu.be/epgwuzzDHsQ)

So far, I have only developed models for four different functions but it's possible to visualise approximations for many more which is why I created this issue. If you're interested, I suggest choosing a function which has not been visualised yet (e.g. tan x), duplicating one of the existing modelling scripts and then rewriting the coefficient calculating function (e.g. `cosineMaclaurinSeriesCoefficients()`) to return the appropriate Maclaurin Series coefficients.

The key to this is finding a pattern in n-derivatives of the function. For example, the derivatives of a Cosine function iterate in cycles of four (1, 0, -1, 0).

The mathematics can be tricky but it's good fun and the programming shouldn't be too hard. I hope you enjoy and the best of luck anyway!

**Comment by ckmganesh** [comment]
Thanks for the video.
I got the maths behind it .So the objective is to find the maclaurins coefficients for as much functions as possible if I'm not wrong. I'll start working on this issue . It'll be helpful if you specify the list of functions fo which you want to find the coefficients. Please let me know if I got something wrong here

**Comment by HarveyRDonnelly** [comment]
Yes well the aim is to visualise the Maclaurin Series approximations approaching the actual function as the polynomial degree is increased. Therefore, the function must calculate the appropriate array of coefficients for a given polynomial degree such that the function is iterative.

I can recommend a few that I was planning to add when I have some time:
- _Hyperbolic Trigonometric Functions (cosh x, sinh x, tanh x)_;
- _Inverse Trigonometric Functions (arcsin x, arccos x, arctan x)_;
- _Binomial Series_;
- _Variations of Exponential/Logarithmic Functions_;

The project is just for a bit of fun, aiming to visualise some beautiful mathematics so you can choose any function you feel would be interesting to model!

**Comment by ckmganesh** [comment]
Yeah I'll try to implement what I can

**Comment by ckmganesh** [comment]
Also regarding the animations part I'm facing a difficulty while executing it

**Comment by ckmganesh** [comment]
![IMG20210114175112.jpg](https://user-images.githubusercontent.com/69971780/104590289-1d7e5f80-5691-11eb-9d8f-62bbb50a9dcc.jpg)

The animations remains the same throughout the whole cycle

Must something be changed for this feature?

**Comment by HarveyRDonnelly** [comment]
Awesome, thanks for contributing!

What exactly is the difficulty with the animation?I haven't tested the code on any machine other than my own.

**Comment by HarveyRDonnelly** [comment]
Assuming that you haven't changed any code, that is strange. Would you mind sending the terminal output?

**Comment by HarveyRDonnelly** [comment]
Also may I ask, does the slider work properly?

**Comment by ckmganesh** [comment]
![IMG20210114175456.jpg](https://user-images.githubusercontent.com/69971780/104590654-a5646980-5691-11eb-98b1-2331983f83fc.jpg)


This is what the terminal says

**Comment by ckmganesh** [comment]
> Also may I ask, does the slider work properly?

No it's not
I tried working it on gitpod but I'm not able to see the output there

**Comment by HarveyRDonnelly** [comment]
Thank you, this may be worth opening a new issue as it's likely due to environment incompatibility. I have only tested this on my own system with Mac OS 10.15.7 and Python 3.7.3.

May I ask what version of Python you are using? I will try my best to debug although it may be difficult using my machine. My apologies for the inconvenience.

**Comment by ckmganesh** [comment]
Windows 10-home 64 bit 
Python 3.7.6

**Comment by HarveyRDonnelly** [comment]
I have raised this as a new issue [here](https://github.com/HarveyRDonnelly/maclaurin-series-models/issues/2). Please add any further details of the problem which I haven't mentioned. I have attached a recording of the software working on my machine for reference.

**Comment by ckmganesh** [comment]
Thanks Harvey.
I'll just work on the coefficient part anyways. I'll update as and when I finish for a function
